### ISSUE TITLE
Can't access titles in Trakt Liked Lists from Private Trakt accounts

### ISSUE THREAD

**Issue by daherlihy** [Tikipeter/tikipeter.github.io#24]
I've liked a number of Lists from other Trakt users and have discovered today that although all the lists appear inside the **Trakt Liked Lists** menu inside FenLight, the titles within each list only appear inside each list if the author Trakt account is not set to Private.

For instance, I can view the titles inside this popular Trakt list: https://trakt.tv/users/fresnobiker/lists/new-hd-releases?sort=added,asc
But on this test list (which I've even set to public) that I created on my private Trakt account, the title(s) I've added aren't displaying when I open the list: https://trakt.tv/users/daherlihy/lists/test-list-public

In short, would it be possible to enable the viewing titles inside liked lists from private Trakt users?

**Comment by daherlihy** [comment]
Closed in error - re-opening as titles still can't be access in liked Trakt lists (or even my own Trakt lists) because the Trakt accounts are set to Private.

**Comment by Tikipeter** [comment]
I'll look into this when I have the time.

In saying that, should it be possible for a user to access the contents of another user's Trakt lists if they are set to private?

What is the point of setting the list to private in this regard?

**Comment by daherlihy** [comment]
The point of someone setting their Trakt account to private is to hide their data from being publicly visible. 

In saying that, a user should (still) be able to access **their own lists** on their own Trakt account once they're logged in to it, regardless of whether their Trakt account is Public or Private. They are their own lists after all that they want to access. Trakt link to own lists: https://trakt.tv/users/[TraktUsername]/lists

Regarding **liked lists** on other private Trakt accounts, consider that the user has been able to **like** them, meaning that they have been granted access to them by the private account(s) in question. For instance, a list can be liked after the URL for it is shared to follower account, or liked after being made accessible between friend accounts (i.e. 2 Trakt accounts which follow each other). Either way, once the list is liked, it is accessible through the Trakt website, hence should also be accessible elsewhere where ever the user has logged into their Trakt account. Trakt link to liked lists: https://trakt.tv/users/[TraktUsername]/lists/liked

Sorry for the long post - hope it makes more sense. Happy to explain further if necessary :)

**Comment by Tikipeter** [comment]
Thanks for the explanation.

Does the "following" of the Trakt accounts have to be reciprocal before a private list is exposed?

For example, if I decide to "Follow" your account on Trakt, do I then have access to your Private Lists, or does that access depend on you also "Following" my account?

Or is access granted on a per-follower basis, manually by the list owner?

**Comment by daherlihy** [comment]
Good question - happy to explain. For a user to Follow a private account on Trakt, they must first click on Follow while viewing that user, which then emails a Follow request to the private account for approval/denial.

If the private account approves it, then the user then has access to the Private accounts watchlist and favourite list on Trakt. For the lists created on the private account, visibility to the user will depend on each list's privacy setting, where there are 4 off:
![image](https://github.com/Tikipeter/tikipeter.github.io/assets/25244119/0a26a658-6465-4fb6-9ac1-39c733fc0eab)

1. **Public** means the list will be visible alongside the watchlist and favourites
2. **Friends** means it's visible only to Trakt accounts which follow each other
3. **Private** means it can't be accessed under any circumstances (even by friends)
4. **Link** means it can only be accessed if a link to it is shared with the follower (otherwise it remains private)

Just to point out that none of these lists that are visible to and accessible by the user are considered as Liked lists until the user actually clicks Like on them (the thumbs up icon).

So in short, for any of the above Lists to be liked by a Private account, there will be an approval process first with the follower and then the follower will still only have access to lists as permitted by the Private account.

Note that all of the above applies also to Public accounts, just without the Follow approval process.

**Comment by Tikipeter** [comment]
I appreciate the time taken to explain.

The POV dev reached out to me the other day regarding this, possibly seeing this issue and recently dealing with this himself.

He outlined the steps taken to remedy the situation for him. I was hoping to not have to do something similar, but probably I will.

The thing is my lists on my Trakt account are set to "Private", yet I can view them fine through Fen Light.

I also just now set my account to "Private" and I can still access all my personal lists with both my account and my lists set to "Private".

**Comment by daherlihy** [comment]
> The thing is my lists on my Trakt account are set to "Private", yet I can view them fine through Fen Light.
> 
> I also just now set my account to "Private" and I can still access all my personal lists with both my account and my lists set to "Private".
Yep there's no issue with access our own lists.

This ticket is just related to accessing **Liked Lists** on Trakt accounts set to Private.

**Comment by Tikipeter** [comment]
Ah OK. This threw me slightly off the scent.......

>In saying that, a user should (still) be able to access their own lists on their own Trakt account once they're logged in to it, regardless of whether their Trakt account is Public or Private. They are their own lists after all that they want to access.

I've been busy sorting out the watched status issues I've had since the last update, but I did manage to at least take a fleeting glance at what I was filtering out with regard to privacy status.

Has the latest version `1.0.45` changed anything with regard to this? If not, then I'll likely need to make bigger changes more in line with what POV dev was explaining.

**Comment by daherlihy** [comment]
Unfortunately not no. No worries if you need more time.

NOTE - I've just noticed that I can no longer access Liked Lists with the Friends privacy setting on Private Trakt accounts on POV either, so its dev must have changed something (I could previously). No problem accessing a Liked List with the Public privacy setting applied (also on the same Private Trakt account). I'll raise that issue with the POV dev.
